### PR TITLE
add err define of mysql to moerr

### PR DESCRIPTION
**Pull request by iamlinjunhong**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3752 

## What this PR does / why we need it:

### CODE REVIEW

**Review by aressu1985 — APPROVED** [review]

**Comment by xzxiong** [comment]
@iamlinjunhong questions:
1. how to maintain mapping: moerr -> mysqlError ? there is so much mysql Error, find the corresponding error is not easy.  例如："No database selected", 我每次用都要扫描整个 mysql_error_define.go (1MB）来确定用哪一个错误码？特别是我们可能就维护 100+个 mysql错误码的映射
3. Should we void some segment ? `1~1000` not recommend already has meaning. like 404: it means page not found.

File: pkg/common/moerr/error.go [hunk 1]
```diff
@@ -32,71 +32,50 @@ const (
...
-	ErrRangeNotIncreasing
-	ErrPartitionsMustBeDefined
-	ErrPartitionColumnList
-	ErrWrongTypeColumnValue
-	ErrUniqueKeyNeedAllFieldsInPf
-	ErrMultipleDefConstInListPart
-	ErrPartitionWrongNoPart
+	ErrNoAvailableBackend = 504
 
 	// Group 10: txn
```
Comment (fengttt) [inline pkg/common/moerr/error.go:66]: Group 6

File: pkg/common/moerr/error.go [hunk 2]
```diff
@@ -32,71 +32,50 @@ const (
 	WARN    = 2
 
 	// Group 1: Internal errors
-	ERROR_START              = 1000
-	INTERNAL_ERROR           = 1001
-	NYI                      = 1002
-	ERROR_FUNCTION_PARAMETER = 1003
+	ERROR_START              = 100
```
Comment (fengttt) [inline pkg/common/moerr/error.go:35]: Do you want to give it a type MOErrorCode to these constants?

**Comment by iamlinjunhong** [comment]
> 1. mysql_error_define

1.The product side needs to confirm which error codes need mysql. It may be deleted later. mysql_ error_ Define is a loaded hash table. You don't need to scan the whole file every time.

2.You're right. I'll change the Mo customization error to 20000+

File: pkg/common/moerr/error.go [hunk 3]
```diff
@@ -108,109 +87,112 @@ type moErrorMsgItem struct {
...
+
 	_ = errors.WithContextWithDepth(ctx, err, 2)
 	return err
 }
 
 type Error struct {
 	MysqlErrCode uint16
 	ErrorCode    MOErrorCode
-	Code         int32
+	Code         uint16
```
Comment (fengttt) [inline pkg/common/moerr/error.go:193]: What is code?   Is it ErrorCode?   

Please add comments to these fields.

**Review by fengttt — COMMENTED** [review]
Generate good, thank you.

Minor comments inline.

**Comment by iamlinjunhong** [comment]
> @iamlinjunhong questions:
> 
> 1. how to maintain mapping: moerr -> mysqlError ? there is so much mysql Error, find the corresponding error is not easy.  例如："No database selected", 我每次用都要扫描整个 mysql_error_define.go (1MB）来确定用哪一个错误码？特别是我们可能就维护 100+个 mysql错误码的映射
> 2. Should we void some segment ? `1~1000` not recommend already has meaning. like 404: it means page not found.



1.The product side needs to confirm which error codes need mysql. It may be deleted later. mysql_ error_ Define is a loaded hash table. You don't need to scan the whole file every time.

2.You're right. I'll change the Mo customization error to 20000+

File: pkg/common/moerr/error.go (hunk 1 above)
Comment (iamlinjunhong) [reply pkg/common/moerr/error.go]: done

File: pkg/common/moerr/error.go (hunk 2 above)
Comment (iamlinjunhong) [reply pkg/common/moerr/error.go]: done

File: pkg/common/moerr/error.go (hunk 3 above)
Comment (iamlinjunhong) [reply pkg/common/moerr/error.go]: Someone defines. I think it can keep only one Code, and done.

**Review by ouyuanning — APPROVED** [review]

File: pkg/frontend/load.go [hunk 4]
```diff
@@ -455,8 +455,8 @@ func collectWriteBatchResult(handler *ParseLineHandler, wh *WriteBatchHandler, e
 	wh.simdCsvErr = nil
 }
 
-func makeParsedFailedError(tp, field, column string, line uint64, offset int) *MysqlError {
-	return NewMysqlError(ER_TRUNCATED_WRONG_VALUE_FOR_FIELD,
+func makeParsedFailedError(tp, field, column string, line uint64, offset int) *moerr.MysqlError {
+	return moerr.NewMysqlError(moerr.ER_TRUNCATED_WRONG_VALUE_FOR_FIELD,
```
Comment (xzxiong) [inline pkg/frontend/load.go:459]: should we just use `moerr.New(...)` ?

Comment (xzxiong) [reply pkg/frontend/load.go:459]: no need to keep `moerr.NewMysqlError`, is that right?

**Comment by xzxiong** [comment]
> > @iamlinjunhong questions:
> > 
> > 1. how to maintain mapping: moerr -> mysqlError ? there is so much mysql Error, find the corresponding error is not easy.  例如："No database selected", 我每次用都要扫描整个 mysql_error_define.go (1MB）来确定用哪一个错误码？特别是我们可能就维护 100+个 mysql错误码的映射
> > 2. Should we void some segment ? `1~1000` not recommend already has meaning. like 404: it means page not found.
> 
> 1.The product side needs to confirm which error codes need mysql. It may be deleted later. mysql_ error_ Define is a loaded hash table. You don't need to scan the whole file every time.
> 
> 2.You're right. I'll change the Mo customization error to 20000+

how to help coder to find his mysql ErrorCode ? `mysql_error_define.go` is so big.

**Comment by iamlinjunhong** [comment]
> > > @iamlinjunhong questions:
> > > 
> > > 1. how to maintain mapping: moerr -> mysqlError ? there is so much mysql Error, find the corresponding error is not easy.  例如："No database selected", 我每次用都要扫描整个 mysql_error_define.go (1MB）来确定用哪一个错误码？特别是我们可能就维护 100+个 mysql错误码的映射
> > > 2. Should we void some segment ? `1~1000` not recommend already has meaning. like 404: it means page not found.
> > 
> > 
> > 1.The product side needs to confirm which error codes need mysql. It may be deleted later. mysql_ error_ Define is a loaded hash table. You don't need to scan the whole file every time.
> > 2.You're right. I'll change the Mo customization error to 20000+
> 
> how to help coder to find his mysql ErrorCode ? `mysql_error_define.go` is so big.

1. Use keywords
2. Reproduce the corresponding error in mysql, and then find the error code
3. With the joint efforts of develop and products, MySQL error codes that we will never use will be deleted.

File: pkg/frontend/load.go (hunk 4 above)
Comment (iamlinjunhong) [reply pkg/frontend/load.go]: done

**Review by fengttt — APPROVED** [review]

**Pull request merged**